src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,61 @@
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  status: string;
+  query: string;
+  onStatusChange: (value: string) => void;
+  onQuerryChange: (value: string) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  status,
+  query,
+  onStatusChange,
+  onQuerryChange,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={e => onStatusChange(e.target.value)}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={e => onQuerryChange(e.target.value)}
+        />
+        {query.length !== 0 ? (
+          <span className="icon is-left">
+            <i className="fas fa-magnifying-glass" />
+          </span>
+        ) : null}
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          {query.length !== 0 ? (
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => onQuerryChange('')}
```

Comment:
The `onClick` handler uses the misspelled `onQuerryChange` prop. Update it to `onQueryChange` to ensure it matches the corrected prop name.